snippet-extractor-output/snippets.yaml

```diff
@@ -1363,6 +1411,40 @@
         dataLabel.text = dataLabelInfo[i].news;
         dataLabel.set(labelProperties[i]);
       }
+      await context.sync();
+    });
+'Excel.ChartSeries:class':
+  - >-
+    // Link to full sample:
+    https://raw.githubusercontent.com/OfficeDev/office-js-snippets/prod/samples/excel/10-chart/chart-leader-lines.yaml
+
+
+    // This function adds data labels to the chart and positions them to
+    demonstrate leader lines.
+
+    await Excel.run(async (context) => {
+      const sheet = context.workbook.worksheets.getItem(sheetName);
+      const chart = sheet.charts.getItemAt(0);
+      const series = chart.series.getItemAt(0);
+      
+      // Enable data labels for the series. Leader lines are enabled by default.
+      series.hasDataLabels = true;
+      series.points.load("items");
+      await context.sync();
+      
+      // Load the top position for each data label.
+      series.points.items.forEach((point) => point.dataLabel.load("top"));
+      await context.sync();
+
+      // Move some data labels to create distance from their chart points.
+      series.points.items[1].dataLabel.top = series.points.items[1].dataLabel.top - 50;
```

Comment:
This would be a little more readable.
```typescript
const point1 = series.points.items[1];
const point2 = series.points.items[2];

point1.dataLabel.top -= 50;
point2.dataLabel.top += 50;
```